Exiled.API/Structs/AttachmentIdentifier.cs

```diff
@@ -0,0 +1,248 @@
+        public static bool operator ==(AttachmentIdentifier left, AttachmentIdentifier right) => left.Name == right.Name && left.Code == right.Code && left.Slot == right.Slot;
+
+        /// <summary>
+        /// Compares two operands: <see cref="AttachmentIdentifier"/> and <see cref="AttachmentIdentifier"/>.
+        /// </summary>
+        /// <param name="left">The left-hand <see cref="AttachmentIdentifier"/> operand to compare.</param>
+        /// <param name="right">The right-hand <see cref="AttachmentIdentifier"/> to compare.</param>
+        /// <returns><see langword="true"/> if the values are equal.</returns>
+        public static bool operator !=(AttachmentIdentifier left, AttachmentIdentifier right) => left.Name != right.Name && left.Code != right.Code && left.Slot != right.Slot;
+
+        /// <summary>
+        /// Compares two operands: <see cref="AttachmentIdentifier"/> and <see cref="FirearmAttachment"/>.
+        /// </summary>
+        /// <param name="left">The <see cref="AttachmentIdentifier"/> to compare.</param>
+        /// <param name="right">The <see cref="FirearmAttachment"/> to compare.</param>
+        /// <returns><see langword="true"/> if the values are equal.</returns>
+        public static bool operator ==(AttachmentIdentifier left, FirearmAttachment right) => left.Name == right.Name && left.Slot == right.Slot;
+
+        /// <summary>
+        /// Compares two operands: <see cref="AttachmentIdentifier"/> and <see cref="FirearmAttachment"/>.
+        /// </summary>
+        /// <param name="left">The <see cref="AttachmentIdentifier"/> to compare.</param>
+        /// <param name="right">The <see cref="FirearmAttachment"/> to compare.</param>
+        /// <returns><see langword="true"/> if the values are not equal.</returns>
+        public static bool operator !=(AttachmentIdentifier left, FirearmAttachment right) => left.Name != right.Name || left.Slot != right.Slot;
+
+        /// <summary>
+        /// Compares two operands: <see cref="AttachmentIdentifier"/> and <see cref="FirearmAttachment"/>.
+        /// </summary>
+        /// <param name="left">The <see cref="FirearmAttachment"/> to compare.</param>
+        /// <param name="right">The <see cref="AttachmentIdentifier"/> to compare.</param>
+        /// <returns><see langword="true"/> if the values are equal.</returns>
+        public static bool operator ==(FirearmAttachment left, AttachmentIdentifier right) => right == left;
+
+        /// <summary>
+        /// Compares two operands: <see cref="AttachmentIdentifier"/> and <see cref="FirearmAttachment"/>.
+        /// </summary>
+        /// <param name="left">The <see cref="FirearmAttachment"/> to compare.</param>
+        /// <param name="right">The <see cref="AttachmentIdentifier"/> to compare.</param>
+        /// <returns><see langword="true"/> if the values are not equal.</returns>
+        public static bool operator !=(FirearmAttachment left, AttachmentIdentifier right) => right != left;
```

Comment:
Won't these always be not equal?

Reply:
`left.Name != right.Name || left.Slot != right.Slot;` You sure?

Reply:
I looked through and wasn't able to see FirearmAttachment or AttachmentIdentifier being related to each other, but regardless, what's the point of even adding this?

Reply:
There's a `right != left` for it, so yea, it's related.
The point of adding thing is that you can check different types without checking each param/field/property every time, and it's a matter of integration btw.